txscript/data/script_tests.json

```diff
@@ -551,6 +556,370 @@
+["", "2DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2DUP requires two inputs"],
+["PUSHDATA4 0x01400000 0x62{16385}", "2DUP 1", "NONE", "ERR_SCRIPT_SIZE", "Exceed max allowed script size"],
+["PUSHDATA4 0x01200000 0x62{8193}", "2DUP 1", "NONE", "ERR_PUSH_SIZE", "Exceed max allowed input size"],
+
+["3DUP test coverge"],
+["0 0 1", "3DUP", "NONE", "OK"],
+["OP_1 OP_1 OP_1", "3DUP ADD ADD ADD ADD ADD 6 EQUAL", "NONE", "OK"],
+["-1 0 1 2", "3DUP DEPTH 7 EQUALVERIFY ADD ADD 3 EQUALVERIFY 2DROP 0 EQUALVERIFY", "NONE", "OK"],
+["NOP", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1 2", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1 1", "3DUP", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["DATA_7 0x7a{7} DATA_15 0x7a{15}", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "3DUP requires three inputs"],
+["", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "3DUP requires three inputs"],
+["1 2 3 4 5 <3DUP>{173}", "1 2 3 4 5 6 <3DUP>{165}", "NONE", "ERR_STACK_SIZE", ">1,024 stack size"],
+["1 2 3 4 5 <3DUP>{172}", "1 TOALTSTACK 2 TOALTSTACK 3 4 5 6 <3DUP>{166}", "NONE", "ERR_STACK_SIZE", ">1,024 stack+altstack size"],
+["PUSHDATA4 0x01400000 0x62{16385}", "3DUP 1", "NONE", "ERR_SCRIPT_SIZE", "Exceed max allowed script size"],
+["PUSHDATA4 0x01200000 0x62{8193}", "3DUP 1", "NONE", "ERR_PUSH_SIZE", "Exceed max allowed input size"],
+
+["2OVER test coverge"],
+["0 1 0 0", "2OVER", "NONE", "OK"],
+["1 2 3 5", "2OVER ADD ADD 8 EQUALVERIFY ADD ADD 6 EQUAL", "NONE", "OK"],
+["NOP", "2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1", "2 3 2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1 1 1", "2OVER", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["0 1 0", "2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2OVER requires two inputs"],
+["", "2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2OVER requires two inputs"],
+["'z'{2049} 'a'{8} 0 0", "2OVER", "NONE", "ERR_PUSH_SIZE", "Exceed max allowed input size"],
+
+["2ROT test coverge"],
+["0 1 0 0 0 0", "2ROT", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 24 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT DROP 25 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 2DROP 20 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 2DROP DROP 21 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 2DROP 2DROP 22 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 2DROP 2DROP DROP 23 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 2ROT 22 EQUAL", "NONE", "OK"],
+["25 24 23 22 21 20", "2ROT 2ROT 2ROT 20 EQUAL", "NONE", "OK"],
+["1 1 1 1 1", "2ROT", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["0 1 0 0", "2ROT 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2ROT requires six inputs"],
+["0 1", "2ROT 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2ROT requires six inputs"],
+["", "2ROT 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2ROT requires six inputs"],
```

Comment:
All of these tests only operate on numeric data, however, the `2ROT` opcode works with arbitrary data, so please add tests which work with non-numeric data as well.  They can follow the `["25 24 23 22 21 20", ...]` series of tests, but using non-numeric data pushes instead.

Also, they should use `EQUALVERIFY` and the necessary mix of `3DROP/2DROP/DROP` opcodes to consume all remaining data one the stack after the `EQUALVERIFY` followed by `1` so they adhere to the `CLEANSTACK` requirements.  In fact, please add that requirement to all of the expected passing tests you add to ensure that's the case as the original issue requested.